### PR TITLE
Fix `cudf::column` constructor args

### PR DESCRIPTION
**Pull request by trxcllnt**
## Description
Update `cudf::column` constructor args to match the changes in https://github.com/rapidsai/cudf/pull/13341.

Also corrects a minor issue in the docs, closes #1154.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by isVoid — APPROVED** [review]

**Comment by trxcllnt** [comment]
Failing test seems like an unrelated cuML bug.

**Review by harrism — APPROVED** [review]

**Comment by isVoid** [comment]
Tracked by: https://github.com/rapidsai/cuml/pull/5434

**Comment by harrism** [comment]
How do we rerun tests? Is the cuML fix in last night's nightly?

**Comment by harrism** [comment]
> How do we rerun tests? Is the cuML fix in last night's nightly?

FIgured this out. Click on failing action, look for rerun icon in top right (circular arrows).
image

**Comment by bdice** [comment]
I went ahead and merged this since it fixes nightlies, which are needed for 23.06 release testing. cc: @beckernick this should be good to go once builds finish on the merge commit.

**Pull request merged**
